catalog/core/tests/test_views.py

```diff
@@ -207,86 +266,75 @@ def test_contact_view(self):
-            CONTACT_US_URL, query_parameters={'format': 'json'})
-        response = self.get(url)
-        json_data = json.loads(response.data['json'])
-        security_hash = json_data['security_hash'] + 'fake'
-        timestamp = json_data['timestamp']
-        time.sleep(5)
-
-        response = self.post(url, {'name': 'John Watson', 'email': 'john@watson.com',
-                                   'message': 'Sherlock want to use this application.',
-                                   'timestamp': timestamp,
-                                   'security_hash': security_hash,
-                                   'contact_number': ''})
-        self.assertTrue(400, response.status_code)
+        timestamp = float(json_data['timestamp'])
+        user = models(User).example()
+        try:
+            validate_email(user.email)
+            flag = True
+        except ValidationError as e:
+            flag = False
+        # Test for all Valid Fields
+        if flag and (contact.strip() is '' or len(contact.strip()) == 10 ) and \
+                        name.strip() is not '' and message.strip() is not '':
+            response = self.post(url, {'name': name,
+                                       'email': user.email,
+                                       'message': message,
+                                       'timestamp': timestamp,
+                                       'security_hash': security_hash,
+                                       'contact_number': contact})
+            self.assertEqual(200, response.status_code)
+        else:
+            # Test for any Invalid Fields
+            response = self.post(url, {'name': name,
+                                       'email': user.email,
+                                       'message': message,
+                                       'timestamp': timestamp,
+                                       'security_hash': security_hash,
+                                       'contact_number': contact})
+            self.assertEqual(400, response.status_code)
 
-    def test_contact_info_submit_with_honeypot_field(self):
+    # Test for dummy timestamp and security hash
+    @settings(max_examples=50)
+    @given(st.text(),st.text(),st.text())
+    def test_contact_info_with_invalid_data(self,name,message,contact):
+        print("Second Definition")
         url = self.reverse(
             CONTACT_US_URL, query_parameters={'format': 'json'})
         response = self.get(url)
         json_data = json.loads(response.data['json'])
-        security_hash = json_data['security_hash']
-        timestamp = json_data['timestamp']
-
-        time.sleep(5)
-        response = self.post(url, {'name': 'John Watson', 'email': 'john@watson.com',
-                                   'message': 'Sherlock want to use this application.',
+        security_hash = json_data['security_hash'] + 'Fake'
+        timestamp = float(json_data['timestamp']) + 1
+        user = models(User).example()
```

Comment:
`models(User).example()` should be in the given decorator as `GENERATE_USER`. `GENERATE_USER` should look something like

```
DEFAULT_ALPHABET = st.characters(whitelist_categories={'Lu', 'Ll', 'Lt', 'Lm', 'Lo'}, blacklist_characters='\x00')

def text(min_size=1, max_size=20):
                  return st.text(alphabet=DEFAULT_ALPHABET, min_size=min_size, max_size=max_size)

GENERATE_USER = models(User,
                  username=st.just(str(username)),
                  email=st.just(str(username) + "@comses.net"),
                  first_name=text(),
                  last_name=text(),
                  password=st.just('')) 
```

(Adapted from `core.comses.net`)